lib/Driver/ToolChains.cpp

```diff
@@ -951,7 +951,21 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   const Driver &D = getDriver();
   const llvm::Triple &Triple = getTriple();
 
-  InvocationInfo II{"ld"};
+  // Configure the toolchain
+  // By default, use the system ld to link.
+  const char* LD = "ld";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_tools_directory)) {
+
+    StringRef toolchainPath(A->getValue());
```

Comment:
Slightly contrary to the rest of the style.  Usually we use = rather than the constructor.  move constructors mean that this extremely low cost.